src/filters/fs.rs

```diff
@@ -524,3 +525,33 @@ mod tests {
         assert_eq!(buf.capacity(), cap);
     }
 }
+
+fn poll_read_buf<T: AsyncRead, B: BufMut>(
+    io: Pin<&mut T>,
+    cx: &mut Context<'_>,
+    buf: &mut B,
+) -> Poll<io::Result<usize>> {
+    if !buf.has_remaining_mut() {
+        return Poll::Ready(Ok(0));
+    }
+
+    let n = {
+        let dst = buf.bytes_mut();
+        let dst = unsafe { &mut *(dst as *mut _ as *mut [MaybeUninit<u8>]) };
```

Comment:
Maybe add `Safety: `dst` may include uninitialized memory, but `ReadBuf` will never uninitialize memory that has already been initialized.` as comment? (See https://github.com/tokio-rs/tokio/discussions/3121)

Reply:
I just copied this code from tokio-util since I couldn't get the latter to work as a dependency. I'm more or less assuming that tokio-util will be used instead by the time this PR is finalized, although I don't know what the maintainers prefer.

Reply:
`tokio-util` has since been upgraded to `0.5.1` which has `poll_read_buf`. I think it makes sense to use it and isolate the unsafe code there, but then maybe the `fs` filter should be feature gated as it now introduces an extra dependency.
What do you think @seanmonstar ?

Reply:
Thanks for the update @jxs :)